### DISCUSSION TITLE
Query: orphan zettels

### DISCUSSION THREAD

**Discussion by Nadrieril** [srid/neuron#204] [Ideas]
I have a script that creates a reference zettel from a link. I tend to collect more references than I actually remember, and I'd like a way to find reference zettels that have not been used anywhere.
Could there be a query that returns zettels that have no incoming links?

**Comment by srid** [comment]
> Could there be a query that returns zettels that have no incoming links?

You mean from the CLI only, right? Because if you evaluate this (non-cf) query on the graph (_EDIT: actually this is not even possible, as the graph would not have been constructed yet_), then suddenly those results would have one incoming link (the zettel from which the query was run); which makes it semantically unfit.

**Comment by srid** [comment]
I suppose we can have `neuron query --graph` dump the entire Zettelkasten graph (vertices being ID only) as JSON, and then you can process it however you wish.

**Comment by srid** [comment]
Interestingly, I just realized a need for this in another project. So here we go: #206

**Comment by Nadrieril** [comment]
I would like that as a link query too actually if that was possible. It would have to mean "orphan except for links coming from orphan queries"

**Comment by srid** [comment]
After #206, I imagine the following queries can be added to the new `GraphQuery` type:

* `GraphQuery_Orphans :: GraphQuery [Zettel]`
* `GraphQuery_Backlinks :: Connection -> GraphQuery [Zettel]`
* `GraphQuery_Frontlinks :: Connection -> GraphQuery [Zettel]`

A noteworthy distinction here is that - only `ZettelQuery` will be evaluated (to determine connections) when building the graph; so the use of `GraphQuery` should not affect the connections; and will only be used if ever for rendering purposes. 

While this is all technically feasible, I'm curious as to just what other uses are for this feature? Is it only about listing orphans somewhere?

**Comment by srid** [comment]
Aside:

>  link query

We don't have clear terminology on these concepts yet. I wouldn't call it a "link query" anymore - as no linking (as in, connection forming) will happen. Any ideas on how to name these things? We have CLI queries, then zettel query links, and now graph queries (which do not link zettels).

**Comment by Nadrieril** [comment]
My usecase is for building a reading list automatically. I do admit that it's not a very orthodox use of neuron ^^ I struggle to see a more legitimate usecase
